build-carthage.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Usage: run this script from the root of Simplified-iOS repo.
+#
+#     build-carthage.sh <Debug | Release>
+#
+# The parameter indicates which configuration of AudioEngine should be used.
+# The Debug build includes simulator slices, while the Release does not.
+# One parameter must be specified.
+#
+# NOTE: The current Release configuration of the AudioEngine framework
+#       specified in Certificates repo will cause a build failure because it
+#       does not contain any slices for x86_64 (simulator) architectures.
+#
+# Description: This scripts wipes your Carthage folder, checks out and rebuilds
+#              all dependencies.
+
+if [ $# -eq 0 ]; then
+  echo "Please specify which AudioEngine configuration you would like to use:"
+  echo "    $0 [Debug | Release]"
+  exit 1
+fi
+
+# deep clean to avoid any caching issues
+rm -rf ~/Library/Caches/org.carthage.CarthageKit
+rm -rf Carthage
+
+carthage checkout --use-ssh
+
+# Simplified-iOS and NYPLAEToolkit depend on the AudioEngine framework. Normally
+# one would express this dependency by adding this line to Cartfile.resolved:
+#
+#   binary "AudioEngine.json" "6.1.15"
+#
+# We've experienced build issues where Carthage may error out with something like:
+#
+#   A shell task (/usr/bin/xcrun dwarfdump --uuid /Users/xyz/git/NYPLAEToolkit/Carthage/Build/iOS/AudioEngine.framework/AudioEngine) failed with exit code 1:
+#   error: /Users/xyz/git/NYPLAEToolkit/Carthage/Build/iOS/AudioEngine.framework/AudioEngine: Invalid data was encountered while parsing the file
+#
+# E.g. That seems to happen when building with `carthage build --configuration Release`
+#
+# We maintain a fork of Carthage that solved the dwarfdump issue above, but using
+# that will produce the following error:
+#
+#   Invalid archive - Found multiple frameworks with the same unarchiving destination:
+#    	file:///var/folders/..../AudioEngine/Release/AudioEngine.framework/
+#     file:///var/folders/..../AudioEngine/Debug/AudioEngine.framework/
+#       to:
+#     file:///Users/xyz/git/Simplified-iOS/Carthage/Build/iOS/AudioEngine.framework/
+#
+# The problem is that the AudioEngine zip specified in the Certificates repo contains
+# 2 builds of the framework (Debug and Release) but Carthage no longer allows that, per
+# https://github.com/Carthage/Carthage#archive-prebuilt-frameworks-into-one-zip-file .
+# Occasionally Carthage may be still able to build, but it is not certain which
+# build of AudioEngine is actually going to use. To control this better, especially
+# when preparing app Release builds) we are manually managing this dependency by
+# copying the pre-built framework binary into the location Carthage expects it:
+
+cd Carthage
+AE_BUILD_CONFIG=$1
+chmod u+x ../../Certificates/SimplyE/iOS/AudioEngineZipURLExtractor.swift
+AUDIOENGINE_ZIP_URL=$( ../../Certificates/SimplyE/iOS/AudioEngineZipURLExtractor.swift ../../Certificates/SimplyE/iOS/AudioEngine.json )
```

Comment:
I'm not sure it's fair to make an assumption about where a developer has their Certificates repo relative to the Simplified-iOS repo

Reply:
I made that assumption / indication in the README

Reply:
Let me rephrase: I indicated to clone Simplified-iOS and Certificates as siblings as a requirement in the manual steps, so it seems like a correct assumption here. While I could avoid that completely by parametrizing, the error checking involved and extra work etc didn't seem worth the time to me. At the end of the day placing Certificates next to Simplified-iOS is just as good as other placements AFAIK. Let me know if you disagree or if I can resolve this.